src/Http/Http.Abstractions/src/Extensions/HttpResponseWritingExtensions.cs

```diff
@@ -93,9 +93,10 @@ private static async Task StartAndWriteAsyncAwaited(this HttpResponse response,
 
         private static void Write(this HttpResponse response, string text, Encoding encoding)
         {
+            var minimumByteSize = encoding.GetMaxByteCount(1);
```

Comment:
The encoding is almost always Encoding.UTF8 right? It might be better to hard code 6 as the minimumByteSize in that case.